### PR TITLE
Refactor GetRequiredPaymentsString / `masternode winners`

### PR DESCRIPTION
**Pull request by UdjinM6**
Move (and refactor) GetRequiredPaymentsString(s) from masternode-payments.{h,cpp} and governance-classes.{cpp,h} into rpc/masternode.cpp. These functions aren't used anywhere else. Should have no changes in behaviour.

~Based on #3856 atm.~

### CODE REVIEW

File: src/rpc/masternode.cpp [hunk 1]
```diff
@@ -296,31 +328,42 @@ UniValue masternode_winners(const JSONRPCRequest& request)
...
+        obj.pushKV(strprintf("%d", h), strPayments);
+    }
+
+    auto projection = deterministicMNManager->GetListForBlock(pindexTip).GetProjectedMNPayees(20);
+    for (size_t i = 0; i < projection.size(); i++) {
+        int h = nChainTipHeight + 1 + i;
+        std::string strPayments = GetRequiredPaymentsString(h, projection[i]);
+        if (strFilter != "" && strPayments.find(strFilter) == std::string::npos) continue;
+        obj.pushKV(strprintf("%d", h), strPayments);
     }
```
Comment (PastaPastaPasta) [inline src/rpc/masternode.cpp:367]: Can you explain what you're doing in these changes?

Comment (UdjinM6) [reply src/rpc/masternode.cpp:367]: It's basically a simplified version of https://github.com/dashpay/dash/pull/3861/files#diff-d99314985bd89b12d74bb2e64afd73ebbb5483249b862ac5a07e5edba98ce8edL251-L282 with all the unnecessary parts dropped (we know `nEndHeight` is always `nChainTipHeight + 20`).

File: src/rpc/masternode.cpp [hunk 2]
```diff
@@ -296,31 +328,42 @@ UniValue masternode_winners(const JSONRPCRequest& request)
...
     UniValue obj(UniValue::VOBJ);
-    auto mapPayments = GetRequiredPaymentsStrings(nHeight - nLast, nHeight + 20);
-    for (const auto &p : mapPayments) {
-        if (strFilter != "" && p.second.find(strFilter) == std::string::npos) continue;
-        obj.pushKV(strprintf("%d", p.first), p.second);
+
+    int nChainTipHeight = pindexTip->nHeight;
+    int nStartHeight = std::max(nChainTipHeight - nCount, 1);
+
+    for(int h = nStartHeight; h <= nChainTipHeight; h++) {
```
Comment (PastaPastaPasta) [inline src/rpc/masternode.cpp:354]:
```suggestion
    for (int h = nStartHeight; h <= nChainTipHeight; h++) {
```

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
One nit: I see it's just a move only, but I think it's okay to fix this formatting

**Review by PastaPastaPasta — APPROVED** [review]
utACK, LGTM

**Review by xdustinface — APPROVED** [review]
ACK

**Pull request merged**